changelog/10865.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix a TypeError in a warning arguments call muted by warnings filter.
```

Comment:
```suggestion
:func:`pytest.warns` now validates that warning object's ``message`` is of type  `str` -- currently in Python it is possible to pass other types than `str` when creating `Warning` instances, however this causes an exception when :func:`warnings.filterwarnings` is used to filter those warnings. See `CPython #103577 <https://github.com/python/cpython/issues/103577>`__ for a discussion.

While this can be considered a bug in CPython, we decided to put guards in pytest as the error message produced without this check in place is confusing.
```

However I'm not sure this should be a `bugfix`, I would argue this is more of an `improvement`.

Reply:
thanks, updated